File: PATTERNS/Pattern32/Pattern32.c
#include <stdio.h>

int main() {
    int n, i, j, k, flag, flag1;
    printf("Enter the input n where row = 2*n - 1: ");
    scanf("%d", &n);
    for (i = 1; i < 2*n; i++) {
        if (i <= n) {
            flag = i;
        } else {
            flag = 2*n - i;
        }
        flag1 = n;
        for (k = flag1; k > 0; k--) {
            printf(" ");
        }
        for (j = flag1; j >= flag; j--) {
            if (j == flag1 || j == flag) {
                printf("* ");
            } else {
                printf("  ");
            }
        }
        printf("\n");
    }
    return 0;
}

File: PATTERNS/Pattern32/Pattern32.go
// *       *
// *     *
// *   *
// * *
// *
// * *
// *   *
// *     *
// *       *

package main

import "fmt"

func main() {
	var n int = 0
	fmt.Println("Enter the number of rows:")
	fmt.Scanf("%d", &n)
	var flag int
	for i := 1; i <= 2*n-1; i++ {
		if i <= n {
			flag = i
		} else {
			flag = 2*n - i
		}
		var flag1 int = n
		for k := flag1; k >= 1; k-- {
			fmt.Printf(" ")
		}
		for j := flag1; j >= flag; j-- {
			if j == flag1 || j == flag {
				fmt.Printf("* ")
			} else {
				fmt.Printf("  ")
			}

		}
		fmt.Printf("\n")
	}
}
